test/e2e/v1beta1/scripts/gh-actions/run-e2e-experiment.py

```diff
@@ -249,6 +255,12 @@ def run_e2e_experiment(
     if experiment.metadata.name == "random":
         MAX_TRIAL_COUNT += 1
         PARALLEL_TRIAL_COUNT += 1
+        if args.trial_pod_annotations:
```

Comment:
Please can we also throw an error if `experiment.spec.trial_template.trial_spec['kind'] != "Job"`.

That will help users of conformance to not define incorrect Trial Spec (e.g. TFJob, PyTorchJob), since we only support `Job` for now.

Reply:
Added `NotImplementedError`